pilot/pkg/model/authorization.go

```diff
@@ -96,44 +108,63 @@ func (policy *AuthorizationPolicies) IsGlobalPermissiveEnabled() bool {
 		policy.rbacConfig.EnforcementMode == rbacproto.EnforcementMode_PERMISSIVE
 }
 
-// GetRolesInNamespace returns ServiceRole in the given namespace.
-func (policy *AuthorizationPolicies) GetRolesInNamespace(ns string) []Config {
+// ListServiceRoles returns ServiceRole in the given namespace.
+func (policy *AuthorizationPolicies) ListServiceRoles(ns string) []Config {
 	if policy == nil {
 		return nil
 	}
 
-	rolesAndBindings := policy.namespaceToV1Policies[ns]
+	rolesAndBindings := policy.namespaceToV1alpha1Policies[ns]
 	if rolesAndBindings == nil {
 		return nil
 	}
 	return rolesAndBindings.Roles
 }
 
-// GetBindingsInNamespace returns the ServiceRoleBindings in the given namespace.
-func (policy *AuthorizationPolicies) GetBindingsInNamespace(ns string) map[string][]*rbacproto.ServiceRoleBinding {
+// ListServiceRoleBindings returns the ServiceRoleBindings in the given namespace.
+func (policy *AuthorizationPolicies) ListServiceRoleBindings(ns string) map[string][]*rbacproto.ServiceRoleBinding {
 	if policy == nil {
 		return map[string][]*rbacproto.ServiceRoleBinding{}
 	}
 
-	rolesAndBindings := policy.namespaceToV1Policies[ns]
+	rolesAndBindings := policy.namespaceToV1alpha1Policies[ns]
 	if rolesAndBindings == nil || rolesAndBindings.Bindings == nil {
 		return map[string][]*rbacproto.ServiceRoleBinding{}
 	}
 
 	return rolesAndBindings.Bindings
 }
 
+// ListAuthorizationPolicies returns the AuthorizationPolicy for the workload in the given namespace.
+func (policy *AuthorizationPolicies) ListAuthorizationPolicies(ns string, workloadLabels map[string]string) []Config {
+	if policy == nil {
+		return nil
+	}
+
+	var ret []Config
+	workload := labels.Instance(workloadLabels)
+	for _, config := range policy.namespaceToV1beta1Policies[ns] {
+		spec := config.Spec.(*authpb.AuthorizationPolicy)
+		selector := labels.Instance(spec.GetSelector().GetMatchLabels())
+		if selector.SubsetOf(workload) {
```

Comment:
Superset or Subset? Or do we just want to see if the two label sets have overlap?

Reply:
I think we should use subset. The following selector 
```yaml
matchLabels:
  "app": "httpbin"
  "version": "v1"
```
should match workload
```yaml
  "app": "httpbin"
  "version": "v1"
  "env": "dev"
```

but not:
```yaml
  "app": "httpbin"
  "version": "v2"
  "env": "dev"
```